middle_end/flambda/simplify/env/rec_uses.mli

```diff
@@ -0,0 +1,83 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*    Pierre Chambart and Guillaume Bury, OCamlPro                        *)
+(*                                                                        *)
+(*   Copyright 2021--2021 OCamlPro SAS                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Rec uses
+
+    This modules aims at tracking use of mainly variables
+    (maybe other things later on ?), with the aim of:
+    - removing unused parameters of *recursive* continuations
+    - moving allocations out of the hot path of recursive
+      continuations (e.g. the allocation of a float that was unboxed
+       by simplify)
+*)
+
+type t
+(** The type tracking the uses of variables (passed through the downwards
+    accumulator.
+    This contains a stack to track in which continuation's handlers
+    the downwards acc currently is. *)
+
+val print : Format.formatter -> t -> unit
+(** Print to a formatter. *)
+
+
+(* {2 Creation and updates} *)
+
+val empty : t
+(** Empty uses *)
+
+val init_toplevel : Continuation.t -> Variable.t list -> t -> t
+(** Initialize the rec_uses so that the stack consists of a single
+    toplevel continuation. *)
+
+val stack_cont : Continuation.t -> Variable.t list -> t -> t
```

Comment:
The implementation shouldn't matter, it should maybe be called `enter_continuation_handler` / `leave_continuation_handler`

Reply:
I have renamed these functions (though using `exit` rather than `leave`, i'm not sure if the difference matters).